src/BaseballGame.js

```diff
@@ -0,0 +1,74 @@
+import { MissionUtils } from "@woowacourse/mission-utils";
+import { generateRandomNum } from "./utils/generateRandomNum";
+import { isInputValid, isEndInputValid } from "./utils/validation";
+import { getResult, printResult } from "./utils/getResult";
+
+class BaseballGame {
+  #COMPUTER_NUM_LEN = 3;
+  #ComputerNums;
+  #UserNums;
+
+  setComputerNums(nums) {
+    this.#ComputerNums = nums;
+  }
+
+  setUserNums(nums) {
+    isInputValid(nums, this.#COMPUTER_NUM_LEN);
+    this.#UserNums = nums.split("").map((str) => Number(str));
+  }
+
+  /**
+   * 게임의 전체 진행을 위한 메서드입니다.
+   */
+  async runGame() {
+    await this.startGame();
+    await this.playGame();
+    await this.endGame();
+  }
+
+  /**
+   * 게임 시작을 위한 메서드입니다.
+   * 게임 시작 멘트를 출력하며, 컴퓨터의 숫자 3개를 정합니다.
+   */
+  async startGame() {
+    await MissionUtils.Console.print("숫자 야구 게임을 시작합니다.");
+    this.setComputerNums(await generateRandomNum(1, 9, this.#COMPUTER_NUM_LEN));
+  }
+
+  /**
+   * 게임 플레이를 위한 메서드입니다.
+   * 값을 입력 받고, 받은 값의 유효성 체크와 결과를 출력합니다.
+   */
+  async playGame() {
+    let isUserWin = false;
```

Comment:
```suggestion
    let isUserWin = false;
```
while 문 중지 조건을 `break`를 사용하지 않으시고 `flag` 방식으로 하신 이유가 있으신가요??

Reply:
가독성을 위해 그렇게 했습니다! 
`break;`로 `while`문을 중지하려면 `while`의 조건 부분에 `(true)`라고 적어야하는데 개인적으로 선호하지 않아서요..

Reply:
오호 그렇군요 좋은 것 같습니다!